src/services/refactors/inlineVariable.ts

```diff
@@ -0,0 +1,194 @@
+    const parent = token.parent;
+
+    // If the node is a variable declaration, make sure it's not in a catch clause or for-loop
+    // and that it has a value.
+    if (isInitializedVariable(parent) && isVariableDeclarationInVariableStatement(parent) && isIdentifier(parent.name)) {
+        // Don't inline the variable if it isn't declared exactly once.
+        if (parent.symbol.declarations?.length !== 1) {
+            return { error: getLocaleSpecificMessage(Diagnostics.Variables_that_share_a_name_with_a_type_or_namespace_in_the_same_scope_cannot_be_inlined) };
+        }
+
+        // Do not inline if the variable is exported.
+        if (findAncestor(parent, node => canHaveModifiers(node) && some(node.modifiers, isExportModifier))) {
+            return undefined;
+        }
+
+        // Find all references to the variable in the current file.
+        const references = getReferenceNodes(parent, checker, file);
+        return references && { references, declaration: parent, replacement: parent.initializer };
+    }
+
+    if (tryWithReferenceToken && isIdentifier(token)) {
+        // Try finding the declaration and nodes to replace via the reference token.
+        const definition = checker.resolveName(token.text, token, SymbolFlags.Value, /*excludeGlobals*/ false);
+        if (definition?.declarations?.length !== 1) {
+            return { error: getLocaleSpecificMessage(Diagnostics.Variables_that_share_a_name_with_a_type_or_namespace_in_the_same_scope_cannot_be_inlined) };
+        }
+
+        const declaration = definition.declarations[0];
+        if (!isInitializedVariable(declaration) || !isVariableDeclarationInVariableStatement(declaration) || !isIdentifier(declaration.name)) {
+            return undefined;
+        }
+
+        const references = getReferenceNodes(declaration, checker, file);
+        return references && { references, declaration, replacement: declaration.initializer };
+    }
+
+    return { error: getLocaleSpecificMessage(Diagnostics.Could_not_find_variable_to_inline) };
+}
+
+function getReferenceNodes(declaration: InitializedVariableDeclaration, checker: TypeChecker, file: SourceFile): Identifier[] | undefined {
+    const references: Identifier[] = [];
+    const cannotInline = FindAllReferences.Core.eachSymbolReferenceInFile(declaration.name as Identifier, checker, file, ref => {
+        // Only inline if all references are reads. Else we might end up with an invalid scenario like:
+        // const y = x++ + 1 -> const y = 2++ + 1
+        if (FindAllReferences.isWriteAccessForReference(ref)) {
+            return true;
+        }
+
+        // typeof needs an identifier, so we can't inline a value in there.
+        if (isTypeQueryNode(ref.parent)) {
+            return true;
+        }
+
+        // Cannot inline recursive declarations (e.g. const foo = () => foo();)
+        if (findAncestor(ref, node => node === declaration.initializer)) {

```

Comment:
I think you can just use `textRangeContainsPositionInclusive` instead of walking up the parents - that way you don't have to walk up the spine.